File: frontend/src/types/coupon.ts
export interface Coupon {
  id: string;
  tenantId: string;
  stripeCouponId?: string;
  name: string;
  type: 'percent_off' | 'amount_off';
  amountOff?: number;
  percentOff?: number;
  currency?: string;
  duration: 'once' | 'repeating' | 'forever';
  durationInMonths?: number;
  maxRedemptions?: number;
  timesRedeemed: number;
  redeemBy?: string;
  isActive: boolean;
  promotionCodes: PromotionCode[];
  metadata?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface PromotionCode {
  id: string;
  couponId: string;
  stripePromotionCodeId?: string;
  code: string;
  isActive: boolean;
  maxRedemptions?: number;
  timesRedeemed: number;
  expiresAt?: string;
  firstTimeTransaction: boolean;
  minimumAmount?: number;
  minimumAmountCurrency?: string;
  createdAt: string;
}

export interface CouponRedemption {
  id: string;
  couponId: string;
  promotionCodeId?: string;
  customerId: string;
  subscriptionId?: string;
  customerName?: string;
  customerEmail?: string;
  amountDiscounted: number;
  currency: string;
  redeemedAt: string;
}

export interface CouponStats {
  totalCoupons: number;
  activeCoupons: number;
  totalRedemptions: number;
  totalDiscountAmount: number;
  mostUsedCouponName?: string;
  redemptionsByMonth: Record<string, number>;
}

export type CouponResponse = Coupon;
export type PromotionCodeResponse = PromotionCode;

export interface CouponFilter {
  page?: number;
  pageSize?: number;
  search?: string;
  type?: string;
  duration?: string;
  isActive?: boolean;
  sortBy?: string;
  sortDirection?: string;
  fromDate?: string;
  toDate?: string;
}

File: backend/Core/Infrastructure/Coupon.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Infrastructure
{
    [Table("Coupons")]
    public class Coupon
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        public Guid TenantId { get; set; }

        [MaxLength(100)]
        public string? StripeCouponId { get; set; }

        [Required, MaxLength(200)]
        public string Name { get; set; } = string.Empty;

        [Required, MaxLength(20)]
        public string Type { get; set; } = "percent_off"; // percent_off | amount_off

        public decimal? AmountOff { get; set; }

        public decimal? PercentOff { get; set; }

        [MaxLength(3)]
        public string? Currency { get; set; }

        [Required, MaxLength(20)]
        public string Duration { get; set; } = "once"; // once | repeating | forever

        public int? DurationInMonths { get; set; }

        public int? MaxRedemptions { get; set; }

        public int TimesRedeemed { get; set; } = 0;

        public DateTime? RedeemBy { get; set; }

        public bool IsActive { get; set; } = true;

        [MaxLength(2000)]
        public string? Metadata { get; set; } // JSON

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        // Navigation
        [ForeignKey("TenantId")]
        public Tenant? Tenant { get; set; }

        public ICollection<PromotionCode> PromotionCodes { get; set; } = new List<PromotionCode>();
        public ICollection<CouponRedemption> Redemptions { get; set; } = new List<CouponRedemption>();
    }
}
